Refreshing reference (#2039)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/core-okta-api/index.md

---
title: Core Okta API
meta:
  - name: description
    content: Learn how the Okta API works and learn about the compatibility rules and design principles.
---
# Core Okta API

> Core Okta API is the primary way that apps and services interact with Okta. You can use it to implement basic auth functions such as sign in your users and programmatically manage your Okta objects. 

Reviewer: ```suggestion
The Core Okta API is the primary way that apps and services interact with Okta. You can use it to implement basic auth functions such as signing in your users and programmatically managing your Okta objects. 
```

## Sign in Your Users 
API endpoints to authenticate your users, challenge for factors, recover passwords, and more. For example:

>   - The [Authentication Api](/docs/reference/api/authn) controls user accesss to Okta.

Reviewer: ```suggestion
  - The [Authentication API](/docs/reference/api/authn) controls user access to Okta.
```

  - The [OpenID Connect & OAuth 2.0 API](/docs/concepts/oauth-openid) controls users access to your applications.

## Manage Okta Objects
REST endpoints to configure objects whenever you need. For example:

> - Use [Apps API](/docs/reference/api/apps/) to manage applications and/or assignments to users or groups.

Reviewer: ```suggestion
- The [Apps API](/docs/reference/api/apps/) is used to manage Apps and their association with Users and Groups.
```

> - See the [Users API](/docs/reference/api/users) for CRUD operations on users.

Reviewer: ```suggestion
- The [Users API](/docs/reference/api/users) is used for CRUD operations on Users.
```

> - Use the [Sessions API](/docs/reference/api/sessions/) to create and manage user's authentication sessions. 

Reviewer: ```suggestion
- The [Sessions API](/docs/reference/api/sessions/) creates and manages user's authentication sessions. 
```

> - Use the [Policy API](/docs/reference/api/policy/) to create and manage settings such as user session lifetime. 

Reviewer: ```suggestion
- The [Policy API](/docs/reference/api/policy/) creates and manages settings such as a user's session lifetime. 
```

> - Use the [Factor API](/docs/reference/api/factors/) to enroll, manage, and verify factors for multifactor authentication (MFA). 

Reviewer: ```suggestion
- The [Factors API](/docs/reference/api/factors/) is used to enroll, manage, and verify factors for multi-factor authentication (MFA). 
```

> <HorizontalDivider/>

Reviewer 2: We don't use horizontal dividers anywhere else in the site. Is there a reason that these were added here?

Author: As I mentioned before, those were present on mockups, so I've added them. 
![image](https://user-images.githubusercontent.com/68992536/116753656-172dac80-aa10-11eb-9a67-979b0c35342b.png)

Do not consume any Okta API unless it is documented on this site. All undocumented endpoints should be considered private, subject to change without notice, and not covered by any agreements.

## Compatibility rules for input parameters
- Requests are compatible irrespective of the order in which the query parameters appear.
- Requests are compatible irrespective of the order in which the properties of the JSON parameters appear.
- New query parameters may be added to future versions of requests.
- Existing query parameters cannot be removed from future versions of requests.
- Existing properties cannot be removed from the JSON parameters in future versions of requests.

## Compatibility rules for JSON responses 
- Responses are compatible irrespective of the order in which the properties appear.
- New properties may be added to future versions of the response.
- Existing properties cannot be removed from future versions of the response.
- Properties with null values may be omitted by responses.

## URL Namespace

> All URLs listed in the documentation should be preceded with your organization's subdomain (tenant) https://{yourOktaDomain}.com/api/{apiversion} and API version.

Reviewer: ```suggestion
All URLs listed in the documentation should be preceded with your organization's subdomain (tenant) and API version: `https://{yourOktaDomain}.com/api/{apiversion}`
```

> The api version is currently v1.

Reviewer: ```suggestion
The API version is currently `v1`.
```
